### PR TITLE
De-duping issue links w/ reference syntax

### LINKED ISSUE #141: Jira bot edits mangle autolinked GitHub references
To reproduce: create a Pull Request that references _both_ a Jira issue, as well as an autolinked reference to another PR. For example, if I create a PR with the following body:
```
This PR addresses [JIRA-001]. See also my_org/my_project#123
```
Then I end up with a PR body in which the Jira issue is correctly linked, but the reference to PR 123 in "my_project" is dropped (i.e., shows up as plain text, not a link).

If I would have created the PR body without the Jira issue link, the autolinked reference would have been just fine.

### LINKED ISSUE #159: Jira bot duplicating ticket links
Similar to #141, the Jira bot is duplicating links to tickets. Please see this gif for a reproduction:

link_inject

### PR DESCRIPTION
**Pull request by dadleyy**
Leveraging [reference links] instead of inline - `[text](link)` - syntax to prevent the issues reported in #159 & #141.

Closes #161 
Closes #159 
Closes #141

| 🎩 | @tcbyrd |
| :--- | :--- |
| 📎 | @returnlytom |

[reference links]: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

### CODE REVIEW

**Comment by tcbyrd** [comment]
Great solution! I'll pull this down and do a full review today. Looks like all the CI failures are related to standardJS, which is an easy fix. Thanks for the suggestion and for opening a PR @dadleyy!

**Comment by returnlytom** [comment]
Nice work @dadleyy, TIL about the `markdown` lib 💡

**Comment by dadleyy** [comment]
thanks guys! @tcbyrd do you need anything from me?

**Comment by dadleyy** [comment]
@tcbyrd can we try to get a fix for this - either this or #161  - shipped asap? My team is very enthused to see this resolved.

**Comment by dadleyy** [comment]
checking in again @tcbyrd - any word on moving this or #161 forward?

File: lib/jira/util/index.js [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
+      const reference = `[${key}]: ${link}`
+
+      if (text.includes(reference)) {
+        continue
+      }
+
+      links.push(reference)
+    }
+
+    return links.length ? [text, links].join('\n\n') : text
```
Comment (tcbyrd) [inline lib/jira/util/index.js]: One minor suggestion here to improve how it renders if you place multiple links in the same comment. Without this it was displaying multiple issues like this:

image


```suggestion
    return links.length ? [text, links.join('\n'].join('\n\n') : text
```

Comment (dadleyy) [reply lib/jira/util/index.js]: ah, thats embarrassing, sorry about that! just pushed a fix + coverage

**Pull request merged**